ipn/local.go

```diff
@@ -386,9 +386,14 @@ func (b *LocalBackend) loadStateWithLock(key StateKey, prefs *Prefs) error {
 	bs, err := b.store.ReadState(key)
 	if err != nil {
 		if err == ErrStateNotExist {
-			b.prefs = NewPrefs()
+			if relaynodePath != "" {
+				b.prefs = *LoadPrefs(relaynodePath, true)
```

Comment:
LoadPrefs returning a pointer but PrefsFromBytes and NewPrefs returning values is weird.

Prefs is big enough and contains pointers/slices that we should just use pointers consistently.

But I think LoadPrefs should return an error and let the caller generate a NewPrefs if the caller wants.

Or it should be renamed LoadOrNewPrefs and document that errors are ignored.

Reply:
Turned Prefs into *Prefs throughout the OSS codebase, and made LoadPrefs return an error.

This is going to cause a fair bit of pain to adjust the corp repo. I haven't inspected it for breakage yet, so there may be some followup amends to fix e.g. e2e tests.

Reply:
I made the prefs thing a distinct cleanup commit, because the effects of the change bubble out quite far and wide, and have nothing to do with legacy prefs loading.